Black/Flake8 formatting and clean up mapping logic

eshant742 opened the pull request:
- Renamed temporary common variable to common_map for clarity
- Unified the self.mapping = …get("custom_mapping", common_map) fallback in __init__()
- Broke long import and function‐call lines into parenthesized, multi‐line blocks with trailing commas (Black style)
- Grouped imports into stdlib, third‑party, and local sections separated by blank lines
- Converted multi‐line ValueError raises into single‐line f‑strings, extracting src into a local variable
- Standardized all method docstrings to single‐line summaries
- Added two blank lines around top‑level class and methods, and ensured a final newline at EOF
- Ran Black and Flake8 against the file—no formatting or lint errors remain
- Added a last line without any spaces
- Formatted the imports correct 
- Removed the extra line which is causing the failed test

kunalsz commented:
@eshant742 To keep the branch history clean, you can squash all the commits into a single one

eshant742 commented:
Yeah sure I will do that

Lorygold commented on buffalogs/impossible_travel/ingestion/ingestion_factory.py:
```diff
@@ -9,42 +9,65 @@
-        self.ingestion_config = config[config["active_ingestion"]]
-        # default mapping: Elasticsearch mapping
-        self.mapping = self.ingestion_config.get("custom_mapping", config["elasticsearch"]["custom_mapping"])
+        active = config["active_ingestion"]
```
it's not necessary to allocate a new variable `active = config["active_ingestion"]`, the `self.active_ingestion = BaseIngestion.SupportedIngestionSources(config["active_ingestion"])` is enough

Lorygold commented on buffalogs/impossible_travel/ingestion/ingestion_factory.py:
```diff
@@ -9,42 +9,65 @@
+        self.ingestion_config = config[active]
+
+        # pull in common mapping if referenced, otherwise use any override
+        common_map = config.get(
```
not really, the `common_map` should be:
* `self.active_ingestion["custom_mapping"]`
Then, each ingestion source could define the value of the `custom_mapping` key in the `ingestion.json` in one of the two possible ways:
- "custom_mapping": "${common_custom_mapping}"
- "custom_mapping": {"key_field1": "value_field1", ...}
so, the self.mapping` variable could be just define as: `self.mapping = self.active_ingestion["custom_mapping"]. Then, the python code should handle with the possibility that the mapping could be a string into which substitute the effective dictionary or directly the dictionary with fields naming

Sounds right to you?

Lorygold commented on buffalogs/impossible_travel/ingestion/ingestion_factory.py:
```diff
@@ -9,42 +9,65 @@
+            "common_custom_mapping",
+            config["elasticsearch"]["custom_mapping"],
+        )
+        for backend in ("elasticsearch", "opensearch", "splunk"):
```
It's not good practice to report the ingestion sources hardcoded here

Lorygold commented:
Hi @eshant742, I have just left some comments on your code.
Additionally, the `ingestion.json` file should be modified, too. It should be something like this:
```
{
    "active_ingestion": "elasticsearch",
    "common_custom_mapping": {
        "@timestamp": "timestamp",
        "_id": "id",
        "_index": "index",
        "user.name": "username",
        "source.ip": "ip",
        "user_agent.original": "agent",
        "source.as.organization.name": "organization",
        "source.geo.country_name": "country",
        "source.geo.location.lat": "lat",
        "source.geo.location.lon": "lon"
    },
    "elasticsearch": {
        "url": "http://elasticsearch:9200/",
        "username": "foobar",
        "password": "bar",
        "timeout": 90,
        "indexes": "cloud-*,fw-proxy-*",
        "bucket_size": 10000,
        "custom_mapping": "${common_custom_mapping}"
    },
    "opensearch": {
        "url": "http://opensearch:9200/",
        "username": "foobar",
        "password": "bar",
        "timeout": 90,
        "indexes": "cloud-*,fw-proxy-*",
        "bucket_size": 10000,
        "custom_mapping": "${common_custom_mapping}"
    },
    "splunk": {
        "host": "splunk",
        "port": 8089,
        "scheme": "http",
        "username": "foobar",
        "password": "bar",
        "timeout": 90,
        "indexes": "main",
        "bucket_size": 10000,
        "custom_mapping": "${common_custom_mapping}"
    }
}
```

eshant742 commented:
Thanks for the feedback @Lorygold  here’s what I’ve done to align with your suggestions:

- Factory: Switched to enum-based lookup and support for "${common_custom_mapping}" or inline dict.
- Base Ingestion: Moved placeholder resolution into the constructor, falling back to an empty dict if none.
- Config: Pulled all field mappings into a top-level common_custom_mapping and replaced per-backend maps with the placeholder.
- Tests: Updated test_initialization to assert factory.mapping equals the shared common_custom_mapping.

Lorygold commented:
Hi @eshant742, some code conflicts afflicts your changes. Could you resolve them, please?
